workflow/controller/operator_test.go

```diff
@@ -3379,6 +3379,395 @@ func TestRetryNodeOutputs(t *testing.T) {
+      - wf-with-pvc-2390440388
+      displayName: '[0]'
+      finishedAt: "2020-10-14T18:12:40Z"
+      id: wf-with-pvc-3090926742
+      name: wf-with-pvc(0)[0]
+      phase: Succeeded
+      progress: 2/2
+      resourcesDuration:
+        cpu: 4
+        memory: 2
+      startedAt: "2020-10-14T18:12:34Z"
+      templateName: entrypoint
+      templateScope: local/wf-with-pvc
+      type: StepGroup
+    wf-with-pvc-3099954303:
+      boundaryID: wf-with-pvc-1973167036
+      displayName: failure
+      finishedAt: "2020-10-14T18:12:44Z"
+      hostNodeName: minikube
+      id: wf-with-pvc-3099954303
+      message: failed with exit code 1
+      name: wf-with-pvc(0)[1].failure
+      outputs:
+        artifacts:
+        - archiveLogs: true
+          name: main-logs
+          s3:
+            accessKeySecret:
+              key: accesskey
+              name: my-minio-cred
+            bucket: my-bucket
+            endpoint: minio:9000
+            insecure: true
+            key: wf-with-pvc/wf-with-pvc-3099954303/main.log
+            secretKeySecret:
+              key: secretkey
+              name: my-minio-cred
+        exitCode: "1"
+      phase: Failed
+      progress: 1/1
+      resourcesDuration:
+        cpu: 2
+        memory: 1
+      startedAt: "2020-10-14T18:12:40Z"
+      templateName: failure
+      templateScope: local/wf-with-pvc
+      type: Pod
+    wf-with-pvc-3143940621:
+      boundaryID: wf-with-pvc-1369025657
+      children:
+      - wf-with-pvc-1083168639
+      displayName: '[0]'
+      finishedAt: "2020-10-14T18:12:50Z"
+      id: wf-with-pvc-3143940621
+      name: wf-with-pvc(1)[0]
+      phase: Succeeded
+      progress: 2/2
+      resourcesDuration:
+        cpu: 3
+        memory: 1
+      startedAt: "2020-10-14T18:12:46Z"
+      templateName: entrypoint
+      templateScope: local/wf-with-pvc
+      type: StepGroup
+    wf-with-pvc-4097333619:
+      boundaryID: wf-with-pvc-1973167036
+      children:
+      - wf-with-pvc-3099954303
+      displayName: '[1]'
+      finishedAt: "2020-10-14T18:12:46Z"
+      id: wf-with-pvc-4097333619
+      message: child 'wf-with-pvc-3099954303' failed
+      name: wf-with-pvc(0)[1]
+      phase: Failed
+      progress: 1/1
+      resourcesDuration:
+        cpu: 2
+        memory: 1
+      startedAt: "2020-10-14T18:12:40Z"
+      templateName: entrypoint
+      templateScope: local/wf-with-pvc
+      type: StepGroup
+  persistentVolumeClaims:
+  - name: data
+    persistentVolumeClaim:
+      claimName: wf-with-pvc-data
+  phase: Running
+  progress: 4/4
+  resourcesDuration:
+    cpu: 7
+    memory: 3
+  startedAt: "2020-10-14T18:12:34Z"
+`
+
+// This test ensures that the PVCs used in the steps are not deleted when
+// the workflow fails
+func TestDeletePVCDoesNotDeletePVCOnFailedWorkflow(t *testing.T) {
+	assert := assert.New(t)
+
+	cancel, controller := newController()
+	defer cancel()
+
+	wfcset := controller.wfclientset.ArgoprojV1alpha1().Workflows("")
```

Comment:
rather than creat wfcset, which is overly complicated and long-winded, can you look for other examples that do `newController(wf)`